File: content-solutions/dp/acorn-houses.js
function solution(acorns) {
  let prev2 = 0;
  let prev1 = 0;
  for (const x of acorns) {
    const cur = Math.max(prev1, prev2 + x);
    prev2 = prev1;
    prev1 = cur;
  }
  return prev1;
}

File: content-solutions/dp/acorn-houses.java
class Solution {
    public int solution(int[] acorns) {
        int prev2 = 0, prev1 = 0;
        for (int x : acorns) {
            int cur = Math.max(prev1, prev2 + x);
            prev2 = prev1;
            prev1 = cur;
        }
        return prev1;
    }
}
